sdb/commands/stacks.py

```diff
@@ -85,47 +65,84 @@ class Stacks(sdb.Command):
+                              __cv_wait_sig+0x15
+                              zthr_procedure+0x51
+                              thread_generic_wrapper+0x74
+                              kthread+0x121
+                              ret_from_fork+0x35
+            ...
+
+        Print stacks containing the l2arc_feed_thread function
+
+            sdb> stacks -c l2arc_feed_thread
+            TASK_STRUCT        STATE             COUNT
+            ==========================================
+            0xffff9521b3f43b80 INTERRUPTIBLE         1
+                              __schedule+0x24e
+                              schedule+0x2c
+                              schedule_timeout+0x15d
+                              __cv_timedwait_common+0xdf
+                              __cv_timedwait_sig+0x16
+                              l2arc_feed_thread+0x66
+                              thread_generic_wrapper+0x74
+                              kthread+0x121
+                              ret_from_fork+0x35
+
+        Print stacks of threads in the RUNNING state
+
+            sdb> stacks -t RUNNING
+            TASK_STRUCT        STATE             COUNT
+            ==========================================
+            0xffff95214ff31dc0 RUNNING               1
+
+        Count the number of stacks in the zfs module
+
+           sdb> stacks -m zfs | count
+           (unsigned long long)12
+
+        Print stacks of the threads started by the zthr command
+
+            sdb> threads | filter obj.comm == "zthr_procedure" | stack
+            TASK_STRUCT        STATE             COUNT
+            ==========================================
+            0xffff9c7e6c268000 INTERRUPTIBLE         5
+                              __schedule+0x24e
+                              schedule+0x2c
+                              cv_wait_common+0x118
+                              __cv_wait_sig+0x15
+                              zthr_procedure+0x45
+                              thread_generic_wrapper+0x74
+                              kthread+0x121
+                              ret_from_fork+0x1f
+
+            0xffff9c7e6c1f8000 INTERRUPTIBLE         1
+                              __schedule+0x24e
+                              schedule+0x2c
+                              schedule_hrtimeout_range_clock+0xb9
+                              schedule_hrtimeout_range+0x13
+                              __cv_timedwait_hires+0x117
+                              cv_timedwait_hires_common+0x4b
+                              cv_timedwait_sig_hires+0x14
+                              zthr_procedure+0x96
+                              thread_generic_wrapper+0x74
+                              kthread+0x121
+                              ret_from_fork+0x1f
+
     """
 
-    names = ["stacks"]
+    names = ["stacks", "stack"]
+    input_type = "struct task_struct *"
+    output_type = "struct task_struct *"
+    mod_start, mod_end = -1, -1
```

Comment:
what would happen if you removed this line and hit the test cases of filtering by module that exists and the bogus module?

Reply:
Do you mean removing the -1 initial values from the variable or removing the variables all together?

Reply:
I see. Removing the declaration entirely and just creating the variables if the module is defined. I tested that and it works for both valid and bogus modules. Does that seem cleaner to you?

Reply:
Note: the lint check doesn't like it you create instance variables on demand. I added an `init` function, initializing the variables there.